jakarta.jdt/org.eclipse.lsp4jakarta.jdt.test/projects/jakarta-sample/src/main/java/io/openliberty/sample/jakarta/jax_rs/NoPublicConstructorProviderClass.java

```diff
@@ -0,0 +1,16 @@
+package io.openliberty.sample.jakarta.jax_rs;
+
+import jakarta.ws.rs.ext.Provider;
+
+@Provider()
+public class NoPublicConstructorProviderClass {
```

Comment:
Is this how a Provider class typically looks? We should try and model this test class as a simple class showcasing how `@Provider` is typically used in practice

See examples shown here: https://eclipse-ee4j.github.io/jakartaee-tutorial/#using-entity-providers-to-map-http-response-and-request-entity-bodies

More examples here (but note that these are using the old javax namespace instead of jakarta): https://www.logicbig.com/how-to/code-snippets/jcode-jax-rs-provider.html